src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndex.cs

```diff
@@ -242,6 +242,61 @@ public override int HandleMap(IndexItem indexItem, IEnumerable mapResults, Lazy<
             }
         }
 
+        private void UpdateMapFields()
+        {
+            if (_isFanout == false)
+                return;
+
+            // we have a fanout in the map reduce index, there are 2 options:
+            // the reduce result has the same OR a different path as one of the reduce keys.
+            // if it's the same path, we need to insert each value separately in the mapped results.
+            // if it isn't, we need to calculate the result for each entry in the fanout (will have the same result).
+            //
+            // Examples:
+            //
+            // The SUM has the same path as one of the reduce keys
+            // from 'Messages'
+            // group by Name, Data[].Items[].ProductName
+            // select Name, sum(Data[].Items[].TotalPrice) as Total
```

Comment:
Is this example correct? `sum()` is defined on `Data[].Items[].TotalPrice` while the group by fields are `Name` and `Data[].Items[].ProductName`

Reply:
That's correct.
I added the `Data[].Items[].ProductName as ProductName` to the result as well.